### PR TITLE
[onert/backend] Add tests for LayerScopeTensorIndex planner

### PR DESCRIPTION
**Pull request by zetwhite**
This PR adds test ceases for LayerScopeTensorIndex based MemoryPlanner.

ONE-DCO-1.0-Signed-off-by: seunghui youn <sseung.youn@samsung.com>

related : https://github.com/Samsung/ONE/pull/13892#pullrequestreview-2278797746

### CODE REVIEW

File: runtime/onert/backend/train/MemoryPlanner.test.cc [hunk 1]
```diff
@@ -276,4 +275,189 @@ TEST(WICPlanner, disposable_claim_release_test)
...
+{
+  PlannerVerifier<BumpPlanner, LayerScopeTensorIndex> p;
+
+  ASSERT_NO_FATAL_FAILURE({
+    p.claim(0, 0, 10, 0);
+    p.claim(1, 0, 20, 10);
+    p.claim(2, 2, 30, 30);
+    p.release(0, 0);
+    p.capacity(60);
+  });
```
Comment (zetwhite) [inline runtime/onert/backend/train/MemoryPlanner.test.cc:288]: Test values are copied from the above disposable tests.

File: runtime/onert/backend/train/MemoryPlanner.test.cc [hunk 2]
```diff
@@ -276,4 +275,189 @@ TEST(WICPlanner, disposable_claim_release_test)
...
+
+    // 7 RELEASE
+    p.release(7, 1);
+
+    // CAPACITY - 125
+    p.capacity(125);
+  });
+}
+
+TEST(FirstFitPlanner, layerscope_neg_release_non_existing_index)
```
Comment (ragmani) [inline runtime/onert/backend/train/MemoryPlanner.test.cc:367]: Negative test names must begin with `neg_` to recognize whether they are negative tests.

File: runtime/onert/backend/train/MemoryPlanner.test.cc [hunk 3]
```diff
@@ -276,4 +275,189 @@ TEST(WICPlanner, disposable_claim_release_test)
...
+
+    // 2 CLAIM - 30
+    p.claim(2, 2, 30, 30);
+
+    // RELEASE non-existing index
+    assert(on_only_debug_mode());
+  });
+}
+
+TEST(FirstFitPlanner, layerscope_neg_release_twice)
```
Comment (ragmani) [inline runtime/onert/backend/train/MemoryPlanner.test.cc:392]: ditto

Comment (zetwhite) [reply runtime/onert/backend/train/MemoryPlanner.test.cc]: :+1: updated

File: runtime/onert/backend/train/MemoryPlanner.test.cc (hunk 2 above)
Comment (ragmani) [reply runtime/onert/backend/train/MemoryPlanner.test.cc]: Could you rename the hidden negative tests above in another PR?

**Review by ragmani — APPROVED** [review]
LGTM

File: runtime/onert/backend/train/MemoryPlanner.test.cc (hunk 2 above)
Comment (zetwhite) [reply runtime/onert/backend/train/MemoryPlanner.test.cc]: Yes, of course. I will fix it now.  :)

**Review by jyoungyun — APPROVED** [review]
LGTM

**Pull request merged**